model_analyzer/triton/client.py

```diff
@@ -0,0 +1,166 @@
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import time
+
+import tritonclient.grpc as grpcclient
+import tritonclient.http as httpclient
+from tritonclient.utils import InferenceServerException
+
+WAIT_FOR_READY_NUM_RETRIES = 100
+
+class TritonClientConfig:
+    def __init__(self):
+        self._client_args = {
+            'url' : None
+        }
+
+    def __getitem__(self, key):
+        return self._client_args[key]
+    
+    def __setitem__(self, key, value):
+        if key in self._client_args:
+            self._client_args[key] = value
+        else:
+            raise Exception("The argument '{}' to the Triton Inference Server"
+                             " is not supported by the model analyzer.".format(key))
+
+class TritonClientFactory:
+    """
+    A factory for creating Triton Client instances
+    """
+    def __init__(self):
+        pass
+
+    def create_client(self, protocol, config):
+        """Create function that dispatches the 
+            correct server implementation based 
+            on the protocol
+
+        Parameters
+        ----------
+        protocol: str
+            The protocol that the client uses to 
+            communicate with the server
+        
+        config: TritonClientCnfig
+        """
+        if protocol is 'grpc':
+            return TritonGRPCClient(config=config)
+        elif protocol is 'http':
+            return TritonHTTPClient(config=config)
+        else:
+            raise Exception("The protocol '{}' for Triton Inference Server"
+                            " is not supported by the model analyzer.".format(protocol))
+
+class TritonClient:
```

Comment:
This is the interface defined for a Triton Client in the original ModelAnalyzer. (load, unload). I also added a wait_for_serve_ready, because I was unsure whether it's best implemented in the server interface or the client interface. I think thats the only thing this file has in common with triton/server.py